aten/src/ATen/native/cpu/layer_norm_kernel.cpp

```diff
@@ -32,15 +36,28 @@ void LayerNormKernelImplInternal(
   T* mean_data = mean->data_ptr<T>();
   T* rstd_data = rstd->data_ptr<T>();
   const T c = T(1) / static_cast<T>(N);
+  constexpr int64_t VEC_SIZE = vec256::Vec256<T>::size();
+  std::array<T, VEC_SIZE> mean_arr;
+  std::array<T, VEC_SIZE> rstd_arr;
   const bool gamma_null = gamma_data == nullptr;
   const bool beta_null = beta_data == nullptr;
   at::parallel_for(0, M, 1, [&](int64_t start, int64_t end) {
     for (int64_t i = start; i < end; ++i) {
       const T* X_ptr = X_data + i * N;
       T* Y_ptr = Y_data + i * N;
-      T mean_val = T(0);
-      T rstd_val = T(0);
-      for (int64_t j = 0; j < N; ++j) {
+      vec256::Vec256<T> mean_vec(0);
+      vec256::Vec256<T> rstd_vec(0);
+      int64_t j = 0;
```

Comment:
Put j declaration inside for-loop?

Reply:
The declaration outside loop can keep the value of j in the next loop (See line 60).